controllers/consoleplugin/consoleplugin_objects.go

```diff
@@ -340,38 +300,97 @@ func (b *builder) metricsService() *corev1.Service {
+	}
+	lconf.Labels = indexFields
+	lconf.TenantID = b.loki.TenantID
+	lconf.ForwardUserToken = b.loki.UseForwardToken()
+	if b.loki.TLS.Enable {
+		if b.loki.TLS.InsecureSkipVerify {
+			lconf.SkipTLS = true
+		} else {
+			caPath := b.volumes.AddCACertificate(&b.loki.TLS, "loki-certs")
+			if caPath != "" {
+				lconf.CAPath = caPath
+			}
+		}
+	}
+	if b.loki.StatusTLS.Enable {
+		if b.loki.StatusTLS.InsecureSkipVerify {
+			lconf.StatusSkipTLS = true
+		} else {
+			statusCaPath, userCertPath, userKeyPath := b.volumes.AddMutualTLSCertificates(&b.loki.StatusTLS, "loki-status-certs")
+			if statusCaPath != "" {
+				lconf.StatusCAPath = statusCaPath
+			}
+			if userCertPath != "" && userKeyPath != "" {
+				lconf.StatusUserCertPath = userCertPath
+				lconf.StatusUserKeyPath = userKeyPath
+			}
+		}
+	}
+	if b.loki.UseHostToken() {
+		lconf.TokenPath = b.volumes.AddToken(constants.PluginName)
+	}
+}
 
-	var features []string
+func (b *builder) setFrontendConfig(fconf *config.FrontendConfig) {
 	if helper.UseEBPF(b.desired) {
 		if helper.IsPktDropEnabled(&b.desired.Agent.EBPF) {
-			features = append(features, "pktDrop")
+			fconf.Features = append(fconf.Features, "pktDrop")
 		}
 
 		if helper.IsDNSTrackingEnabled(&b.desired.Agent.EBPF) {
-			features = append(features, "dnsTracking")
+			fconf.Features = append(fconf.Features, "dnsTracking")
 		}
 
 		if helper.IsFlowRTTEnabled(&b.desired.Agent.EBPF) {
-			features = append(features, "flowRTT")
+			fconf.Features = append(fconf.Features, "flowRTT")
 		}
 	}
+	fconf.RecordTypes = helper.GetRecordTypes(&b.desired.Processor)
+	fconf.PortNaming = b.desired.ConsolePlugin.PortNaming
+	fconf.QuickFilters = b.desired.ConsolePlugin.QuickFilters
+	fconf.AlertNamespaces = []string{b.namespace}
+	fconf.Sampling = helper.GetSampling(b.desired)
+}
 
-	config := map[string]interface{}{
-		"recordTypes":     outputRecordTypes,
-		"portNaming":      b.desired.ConsolePlugin.PortNaming,
-		"quickFilters":    b.desired.ConsolePlugin.QuickFilters,
-		"alertNamespaces": []string{b.namespace},
-		"sampling":        helper.GetSampling(b.desired),
-		"features":        features,
+//go:embed config/static-frontend-config.yaml
+var staticFrontendConfig []byte
+
+// returns a configmap with a digest of its configuration contents, which will be used to
+// detect any configuration change
+func (b *builder) configMap() (*corev1.ConfigMap, string, error) {
+	config := config.PluginConfig{}
+	// configure server
+	config.Server.CertPath = "/var/serving-cert/tls.crt"
+	config.Server.KeyPath = "/var/serving-cert/tls.key"
+
+	// configure loki
+	b.setLokiConfig(&config.Loki)
+
+	// configure frontend from embedded static file
+	err := yaml.Unmarshal(staticFrontendConfig, &config.Frontend)
+	if err != nil {
+		return nil, "", err
 	}
+	b.setFrontendConfig(&config.Frontend)
 
-	configStr := "{}"
-	if bs, err := yaml.Marshal(config); err == nil {
+	var configStr string
+	bs, err := yaml.Marshal(config)
```

Comment:
good thing to add error handling here ... since you did that you could also do the same line 407 below :-) 
(error currently ignored on `hasher.Write` )

Reply:
https://github.com/netobserv/network-observability-operator/pull/477/commits/9e37a34b77e30ded8990f5df1cd31a323fa217d0